scripts/scr_start_load/scr_start_load.gml

```diff
@@ -1,547 +1,122 @@
-                    }
-                    if (obj_ini.veh_role[_companies, _equip] == "Land Raider") {
-                        obj_ini.veh_lid[_companies, _equip] = ship;
-                        obj_ini.veh_wid[_companies, _equip] = 2;
-                        obj_ini.veh_loc[_companies, _equip] = obj_ini.ship_location[ship];
-                        break;
-                    }
-                    if (obj_ini.veh_role[_companies, _equip] == "Land Speeder") {
-                        obj_ini.veh_lid[_companies, _equip] = ship;
-                        obj_ini.veh_wid[_companies, _equip] = 3;
-                        obj_ini.veh_loc[_companies, _equip] = obj_ini.ship_location[ship];
-                        break;
-                    }
-                    if (obj_ini.veh_role[_companies, _equip] == "Whirlwind") {
-                        obj_ini.veh_lid[_companies, _equip] = ship;
-                        obj_ini.veh_wid[_companies, _equip] = 4;
-                        obj_ini.veh_loc[_companies, _equip] = obj_ini.ship_location[ship];
-                        break;
-                    }
-                }
-            }
-        }
-
-        remove_size += company_size;
-        obj_ini.ship_carrying[ship] += company_size;
-        obj_ini.man_size -= company_size;
-        company_size = 0;
-    }
-
-    var first_frigate, next_company, squeeze;
-    next_company = 1;
-    squeeze = 0;
-    first_frigate = ship + 1;
-
-    repeat (20) {
-        next_company += 1;
-        squeeze = 0;
-
-        if (next_company <= 9) {
-            if (next_company == 2) {
-                if (obj_ini.ship_size[1] == 3) {
-                    ship = 1;
-                    squeeze = 1;
-                }
-            }
-
-            if (next_company > 2) {
-                if (companies_loaded == 1 && obj_ini.ship_size[ship] > 2) then squeeze = 1;
-            }
-
-            if (squeeze == 0) {
-                ship += 1;
-                _companies += 1;
-                _equip = 0;
-                company_size = 0;
-                companies_loaded = 1;
-
-                repeat (500) {
-                    _equip += 1;
-  // Check if _companies and _equip are within valid range
-    if (_companies >= 0 && _companies <= 100 && _equip >= 0 && _equip <= 500) {
-        if (obj_ini.role[_companies, _equip] != "") {
-            obj_ini.lid[_companies, _equip] = ship;
-            obj_ini.wid[_companies, _equip] = 0;
-            obj_ini.loc[_companies, _equip] = obj_ini.ship_location[ship];
-            company_size += 1;
+	
+		//loop through companies. try and load whole company onto single ship else spread company across largest ships with remaining space
+	for (var _comp =0; _comp<10;_comp++;){
+		var _company_size = 0;
+		var company_loader =[];//array of companies marines
+		var company_vehicle = [];//array of companies vehicles
+		var  ship_fit = true;
+		for (var _marine =1;_marine<(array_length(obj_ini.role[_comp])-1);_marine++){

```

Comment:
var is named `_marine`, but below it's used in vehicle context: `if (_marine <array_length(obj_ini.veh_role[_comp]))`. I'd recommend to change `_marine` into `_company_object_index` or `_company_asset_index` or something. To make it a bit clearer.

Reply:
oversight from lazy copy and pasting, will fix.